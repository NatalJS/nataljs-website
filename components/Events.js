import React from 'react';

const Events = () => (
  <section id="incoming-events" className="events">
    <h1 className="section-title">Próximos Eventos?</h1>
    <div className="events__container">
      <div className="events__block">
        <div className="col-3 events__time">
          <span>22.09</span>
          <span>8H ÀS 12</span>
        </div>
        <div className="col-9 events__body">
          <div className="events__title">O estado atual do front-end no mercado de trabalho</div>
          <div className="events__location">IMD UFRN</div>
        </div>
      </div>

      <div className="events__block">
        <div className="events__day">
          KEYNOTES
          {' '}<span>8H30 ÀS 9H30</span>
        </div>

        <div className="events__speakers">
          <div className="media event__media event__media--sticker1">
            <img className="event__image" src="/static/images/speakers/avatar-jaon.png" alt="jaon" />
            <div className="media-body">
              <h4 className="event__talk">Javascript para Dinossauros</h4>
              <p className="event__speaker">Jaon <span>Front-End na Evolux</span></p>
            </div>
          </div>
          <div className="media event__media event__media--sticker2">
            <img className="event__image" src="/static/images/speakers/avatar-neto.png" alt="neto" />
            <div className="media-body">
              <h4 className="event__talk">Hyped JS: o que os devs descolados estão usando</h4>
              <p className="event__speaker">Neto Guimarães <span>Front-End na Evolux</span></p>
            </div>
          </div>
          <div className="media event__media event__media--sticker3">
            <img className="event__image" src="/static/images/speakers/avatar-marcell.png" alt="marcell" />
            <div className="media-body">
              <h4 className="event__talk">Conhecendo a Comunidade open source</h4>
              <p className="event__speaker">Marcell Guilherme <span>Front-End na Evolux</span></p>
            </div>
          </div>
        </div>

        <div className="events__day">
          WORKSHOPS
          {' '}<span>10H ÀS 12H</span>
        </div>

        <div className="events__speakers">
          <div className="media event__media event__media--sticker1">
            <img className="event__image" src="/static/images/speakers/avatar-jhonny.png" alt="jhonny" />
            <div className="media-body">
              <h4 className="event__talk">Hello World: criando e configuando seu primeiro projeto web</h4>
              <p className="event__speaker">Jhonny Michel <span>Front-End na Evolux</span></p>
            </div>
          </div>
          <div className="media event__media event__media--sticker2">
            <img className="event__image" src="/static/images/speakers/avatar-gustavo.png" alt="gustavo" />
            <div className="media-body">
              <h4 className="event__talk">Procura de heróis na Marvel com Javascript</h4>
              <p className="event__speaker">Gustavo Freire <span>Front-End na Evolux</span></p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
);

export default Events;
