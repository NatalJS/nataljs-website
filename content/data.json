{
  "events": [
    {
      "id": 2,
      "title": "O estado atual do front-end no mercado de trabalho",
      "address": "IMD UFRN",
      "startAt": "2018-09-15T08:30:00-03:00",
      "subscriptionLink": null,
      "endAt": "2018-08-15T12:00:00-03:00",
      "sections": [
        {
          "id": 4,
          "title": "KEYNOTES",
          "subtitle": "8H30 ÀS 9H30",
          "talks": [
            {
              "id": 7,
              "title": "Javascript para Dinossauros",
              "room": null,
              "speaker": "Jaon",
              "position": "Front-End na Evolux",
              "image": "/static/images/speakers/avatar-jaon.png",
              "startAt": "2018-09-22T09:00:00-03:00"
            },
            {
              "id": 8,
              "title": "Hyped JS: o que os devs descolados estão usando",
              "room": null,
              "speaker": "Neto Guimarães",
              "position": "Front-End na Evolux",
              "image": "/static/images/speakers/avatar-neto.png",
              "startAt": "2018-09-22T09:20:00-03:00"
            },
            {
              "id": 9,
              "title": "Conhecendo a Comunidade open source",
              "room": null,
              "speaker": "Marcell Guilherme",
              "position": "Front-End na Evolux",
              "image": "/static/images/speakers/avatar-marcell.png",
              "startAt": "2018-09-22T09:40:00-03:00"
            }
          ]
        },
        {
          "id": 5,
          "title": "WORKSHOPS",
          "subtitle": "10H ÀS 12H",
          "talks": [
            {
              "id": 10,
              "title": "Procura de heróis na Marvel com Javascript",
              "room": null,
              "speaker": "Gustavo Freire",
              "position": "Front-End na Evolux",
              "image": "/static/images/speakers/avatar-gustavo.png",
              "startAt": "2018-09-22T10:00:00-03:00"
            },
            {
              "id": 11,
              "title": "Hello World: criando e configuando seu primeiro projeto web",
              "room": null,
              "speaker": "Jhonny Michel ",
              "position": "Front-End na Evolux",
              "image": "/static/images/speakers/avatar-jhonny.png",
              "startAt": "2018-09-22T10:00:00-03:00"
            }
          ]
        }
      ]
    },
    {
      "id": 1,
      "title": "Natal JS 2: Javascripters, rejoice!",
      "address": "IMD UFRN",
      "startAt": "2019-05-15T08:30:00-03:00",
      "subscriptionLink": "https://docs.google.com/forms/d/e/1FAIpQLSdWT3MnHCGjhBp7dn5KYTfd61eysoO93TB8Alz12u2vmXiJ-Q/viewform",
      "endAt": "2019-05-15T12:00:00-03:00",
      "sponsors": [
        {
          "id": 1,
          "name": "Evolux Sistemas",
          "image": "/static/images/evolux.svg",
          "link": "https://www.evolux.net.br/vagas"
        }
      ],
      "sections": [
        {
          "id": 1,
          "title": "PALESTRAS: SALA B206",
          "subtitle": "INICIO 8H30",
          "talks": [
            {
              "id": 1,
              "title": "Javascript everywhere",
              "speaker": "Stefan Yohansson",
              "image": "/static/images/speakers/avatar-stefan.png",
              "startAt": "2019-05-15T09:00:00-03:00"
            },
            {
              "id": 2,
              "title": "JavaScript para programadores Python",
              "speaker": "Wallysson Lima",
              "image": "/static/images/speakers/avatar-wallysson.png",
              "startAt": "2019-05-15T09:20:00-03:00"
            },
            {
              "id": 3,
              "title": "React Native autenticação com AWS Amplify",
              "speaker": "Gilmar Silva",
              "image": "/static/images/speakers/avatar-gilmar.png",
              "startAt": "2019-05-15T09:40:00-03:00"
            },
            {
              "id": 4,
              "title": "Desenvolvimento 3D na web",
              "speaker": "Daniel Fernandes",
              "image": "/static/images/speakers/avatar-daniel.png",
              "startAt": "2019-05-15T10:00:00-03:00"
            }
          ]
        },
        {
          "id": 2,
          "title": "COFFEE BREAK",
          "subtitle": "10H25"
        },
        {
          "id": 3,
          "title": "WORKSHOPS",
          "subtitle": "INÍCIO 10H50",
          "description": "Os workshops ocorrerão simultaneamente.",
          "talks": [
            {
              "id": 5,
              "title": "Contribuição para grandes projetos opensource",
              "room": "SALA A303",
              "speaker": "Stefan Yohansson",
              "image": "/static/images/speakers/avatar-stefan.png",
              "startAt": "2019-05-15T10:50:00-03:00"
            },
            {
              "id": 6,
              "title": "Criando sua primeira aplicação em React.js",
              "room": "SALA A104",
              "speaker": "Gustavo Freire",
              "image": "/static/images/speakers/avatar-gustavo.png",
              "startAt": "2019-05-15T10:50:00-03:00"
            }
          ]
        }
      ]
    },
    {
      "id": 3,
      "title": "Natal JS 3: A terceira vez é ainda melhor",
      "address": "IMD UFRN",
      "startAt": "2019-08-31T08:30:00-03:00",
      "subscriptionLink": null,
      "endAt": "2019-08-31T12:00:00-03:00",
      "call4papers": {
        "endAt": "2019-08-23T18:00:00-03:00",
        "link": "https://speakerfight.com/events/iii-meetup-da-nataljs/"
      },
      "sections": [
        {
          "id": 6,
          "title": "PALESTRAS: 1 PARTE",
          "subtitle": "8H30",
          "talks": [
            {
              "id": 12,
              "title": "Além de validar formulários: usando React pra criar um telefone web",
              "speaker": "Marcell G. C. da Silva",
              "image": "/static/images/speakers/avatar-marcell.png",
              "startAt": "2019-08-31T09:00:00-03:00"
            },
            {
              "id": 13,
              "title": "Como NÃO aprender react.js",
              "speaker": "Thayrone Santos",
              "image": "/static/images/speakers/avatar-thayrone.png",
              "startAt": "2019-08-31T09:20:00-03:00"
            },
            {
              "id": 14,
              "title": "Como diminuir a chance do seu código dar M#$%",
              "speaker": "Adalberto Camacho Jr.",
              "image": "/static/images/speakers/avatar-adalberto.png",
              "startAt": "2019-08-31T09:40:00-03:00"
            },
            {
              "id": 15,
              "title": "Sou um QA, só sei Python e quero aprender JS",
              "speaker": "Rodrigo Muniz",
              "image": "/static/images/speakers/avatar-rodrigo.png",
              "startAt": "2019-08-31T10:00:00-03:00"
            }
          ]
        },
        {
          "id": 7,
          "title": "COFFEE BREAK",
          "subtitle": "10H20"
        },
        {
          "id": 8,
          "title": "PALESTRAS: 2 PARTE",
          "subtitle": "10H40",
          "talks": [
            {
              "id": 16,
              "title": "Debugger is on the table: Utilizando Debugger + Dev Tools para investigações mais avançadas",
              "speaker": "Zenobio",
              "image": "/static/images/speakers/avatar-zenobio.png",
              "startAt": "2019-08-31T10:40:00-03:00"
            },
            {
              "id": 17,
              "title": "O processo de não reinventar a roda usando lodash.js",
              "speaker": "Neto Guimarães",
              "image": "/static/images/speakers/avatar-neto.png",
              "startAt": "2019-08-31T11:00:00-03:00"
            },
            {
              "id": 18,
              "title": "Introdução a Static Site Generators e porque GatsbyJS pode ser sua escolha",
              "speaker": "Igor Santos",
              "image": "/static/images/speakers/avatar-igor.png",
              "startAt": "2019-08-31T11:20:00-03:00"
            },
            {
              "id": 19,
              "title": "Javacript para IOT",
              "speaker": "Gustavo Freire Santos",
              "image": "/static/images/speakers/avatar-gustavo.png",
              "startAt": "2019-08-31T11:40:00-03:00"
            }
          ]
        }
      ]
    }
  ]
}
