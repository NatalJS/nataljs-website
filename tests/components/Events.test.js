import renderer from 'react-test-renderer';
import Events from 'components/Events';
import data from '../../content/data.json';

describe('Component Events', () => {
  let Component;

  beforeAll(() => {
    Component = setup(Events, { events: data.events });
  });

  it('should render component correctly', () => {
    const wrapper = renderer.create(Component());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
