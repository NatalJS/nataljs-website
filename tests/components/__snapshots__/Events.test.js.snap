// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Events should render component correctly 1`] = `
<section
  className="events"
  id="incoming-events"
>
  <h1
    className="section-title"
  >
    Próximos Eventos
  </h1>
  <div
    className="events__container"
  >
    <div
      className="events__block"
    >
      <div
        className="col-3 events__time"
      >
        <span>
          08.31
        </span>
        <span>
          08:30H ÀS 12H
        </span>
      </div>
      <div
        className="col-9 events__body"
      >
        <div
          className="events__title"
        >
          Natal JS 3: A terceira vez é ainda melhor
        </div>
        <div
          className="events__location"
        >
          IMD UFRN
        </div>
      </div>
    </div>
    <div
      className="events__block"
    >
      <div
        className="events__day"
      >
        PALESTRAS: 1 PARTE
         
        <span>
          8H30
        </span>
      </div>
      <div
        className="events__speakers"
      >
        <div
          className="media event__media event__media--sticker1"
        >
          <img
            alt="Marcell G. C. da Silva"
            className="event__image"
            src="/static/images/speakers/avatar-marcell.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              09H
            </div>
            <h4
              className="event__talk"
            >
              Além de validar formulários: usando React pra criar um telefone web
            </h4>
            <p
              className="event__speaker"
            >
              Marcell G. C. da Silva
            </p>
          </div>
        </div>
        <div
          className="media event__media event__media--sticker2"
        >
          <img
            alt="Thayrone Santos"
            className="event__image"
            src="/static/images/speakers/avatar-thayrone.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              09:20H
            </div>
            <h4
              className="event__talk"
            >
              Como NÃO aprender react.js
            </h4>
            <p
              className="event__speaker"
            >
              Thayrone Santos
            </p>
          </div>
        </div>
        <div
          className="media event__media event__media--sticker3"
        >
          <img
            alt="Adalberto Camacho Jr."
            className="event__image"
            src="/static/images/speakers/avatar-adalberto.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              09:40H
            </div>
            <h4
              className="event__talk"
            >
              Como diminuir a chance do seu código dar M#$%
            </h4>
            <p
              className="event__speaker"
            >
              Adalberto Camacho Jr.
            </p>
          </div>
        </div>
        <div
          className="media event__media event__media--sticker4"
        >
          <img
            alt="Rodrigo Muniz"
            className="event__image"
            src="/static/images/speakers/avatar-rodrigo.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              10H
            </div>
            <h4
              className="event__talk"
            >
              Sou um QA, só sei Python e quero aprender JS
            </h4>
            <p
              className="event__speaker"
            >
              Rodrigo Muniz
            </p>
          </div>
        </div>
      </div>
      <div
        className="events__day"
      >
        COFFEE BREAK
         
        <span>
          10H20
        </span>
      </div>
      <div
        className="events__day"
      >
        PALESTRAS: 2 PARTE
         
        <span>
          10H40
        </span>
      </div>
      <div
        className="events__speakers"
      >
        <div
          className="media event__media event__media--sticker1"
        >
          <img
            alt="Zenobio"
            className="event__image"
            src="/static/images/speakers/avatar-zenobio.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              10:40H
            </div>
            <h4
              className="event__talk"
            >
              Debugger is on the table: Utilizando Debugger + Dev Tools para investigações mais avançadas
            </h4>
            <p
              className="event__speaker"
            >
              Zenobio
            </p>
          </div>
        </div>
        <div
          className="media event__media event__media--sticker2"
        >
          <img
            alt="Neto Guimarães"
            className="event__image"
            src="/static/images/speakers/avatar-neto.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              11H
            </div>
            <h4
              className="event__talk"
            >
              O processo de não reinventar a roda usando lodash.js
            </h4>
            <p
              className="event__speaker"
            >
              Neto Guimarães
            </p>
          </div>
        </div>
        <div
          className="media event__media event__media--sticker3"
        >
          <img
            alt="Igor Santos"
            className="event__image"
            src="/static/images/speakers/avatar-igor.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              11:20H
            </div>
            <h4
              className="event__talk"
            >
              Introdução a Static Site Generators e porque GatsbyJS pode ser sua escolha
            </h4>
            <p
              className="event__speaker"
            >
              Igor Santos
            </p>
          </div>
        </div>
        <div
          className="media event__media event__media--sticker4"
        >
          <img
            alt="Gustavo Freire Santos"
            className="event__image"
            src="/static/images/speakers/avatar-gustavo.png"
          />
          <div
            className="media-body"
          >
            <div
              className="events__date-time"
            >
              11:40H
            </div>
            <h4
              className="event__talk"
            >
              Javacript para IOT
            </h4>
            <p
              className="event__speaker"
            >
              Gustavo Freire Santos
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
